### DISCUSSION TITLE
为什么我blog的评论区遭到攻击，而且投放的是你们的广告

### DISCUSSION THREAD

**Discussion by Moemu** [vuepress-theme-hope/vuepress-theme-hope#785] [Q&A] [answered]
我的Blog（muspace.top）评论区疑似遭到攻击，在短短两天内产生了1.5k的评论（目前所有评论已经被我删除），请问这和你们的项目是不是有关联，若无关联。为何你们的项目又出现在上面？请你们向我合理解释，否则我将会通过法律方式解决此问题。
![](https://ftp.bmp.ovh/imgs/2021/07/47860659f5577a2e.png)
![](https://i.bmp.ovh/imgs/2021/07/8e06f6e72d2cf07f.png)
![](https://i.bmp.ovh/imgs/2021/07/eb0a7aeef41fcdbe.png)

**Comment by Moemu** [comment, marked as answer]
现在我通过有关人员的简介了解这次事件发生原因，请你们务必在这个项目的readme.md最上方特别表明这次事件发生的原因，感谢支持

**Reply by Moemu** [reply]
请注意我并没有仇视您，而且我通过后台查询到每个评论的浏览器，系统，IP(v4 v6)都不相同，想要发现幕后凶手有点难度，希望您不要太在意

**Reply by Mister-Hope** [reply]
你是在使用 Waline 对吧？我本人就是 Waline 的核心贡献者，我没有道理往自己的做的项目的用户上去灌垃圾评论。🤡

**Reply by Moemu** [reply]
确实，您知不知道这种不是您本人产生的垃圾评论是什么时候开始产生的呢。

**Reply by Mister-Hope** [reply]
另外这个主题的网站、waline 的官方，我和waline另一个核心贡献者的博客也同样遭到类似内容的爆破。同样的爆破在半年前就已经发生过了，那次也是同样的内容，只不过我还不是核心贡献者，只有@lizheming 他的 Waline 项目，爆破地点发生在 Valine 的大量博客上。

不是很清楚，但是攻击的主要对象是 Waline 本身。

**Reply by Mister-Hope** [reply]
攻击者试图通过垃圾评论的方式，同时在非 Waline(主要是 Valine) 和 Waline 评论系统上大量恶意添加我和 lizheming的项目。请稍等我给你找一下lizheming将近一年前发的博客。

**Reply by Moemu** [reply]
嗯，这样吗？现在我Blog的垃圾评论又开始产生了，请问有什么方法能阻止这种现象的发生呢？

**Reply by Mister-Hope** [reply]
我不是很清楚这次事件的起因，也许是那名攻击者再次伪造我们两位核心开发者大量发送垃圾评论，导致某一名用户不满，转而掉头攻击所有 Waline  用户，也许这就是攻击者本人在发动两面攻击，我们目前也没有更多的细节。攻击者的 IP 深度超过2500.

**Reply by Mister-Hope** [reply]
您可以暂时开启评论审计，请添加环境变量 `COMMENT_AUDIT` 并设置为 `TRUE` ，同时关闭邮件通知。

您也可以自行添加 https://waline.js.org/reference/server.html#presave-comment 这个钩子，尽可能对这些垃圾评论进行检测直接丢弃。

记得两者都需要重新部署。

**Reply by Moemu** [reply]
如果那样的话我也无能为力，只能祝福你们未来少受到甚至不受到诸如此类的攻击，感谢您的回复！

**Reply by Mister-Hope** [reply]
这里是之前我们被恶意冒充时，lizheming 的回复： https://imnerd.org/spam-statement.html

**Reply by Mister-Hope** [reply]
Explaination added in https://github.com/vuepress-theme-hope/vuepress-theme-hope/discussions/786

**Reply by Moemu** [reply]
开了审核机制后，发现攻击者评论会自动通过，而且现在还在增加...

**Reply by Mister-Hope** [reply]
你修改环境变量之后需要重新部署的。😉

**Reply by Moemu** [reply]
我早就重新部署了，我还测试了一下，确实开启了

**Reply by Moemu** [reply]
![image](https://user-images.githubusercontent.com/72406624/126135969-fabfb7ee-1a0e-4c96-a341-8fbec325d1e3.png)

**Reply by Moemu** [reply]
为了躲避攻击，现在我只能暂时关闭评论功能

**Reply by Mister-Hope** [reply]
这不是已经待审核了么您顺带把邮件通知暂时关了应该就好了，攻击者看不到效果他过一会儿就停止攻击了。

不过我自己的博客，项目以及waline 官网这两天收到了四波攻击了。目前我自己的项目是用钩子直接屏蔽掉了。

**Reply by Moemu** [reply]
我现在关了评论区，我再说一次，我的评论发出去是待审核的，但是攻击者的评论没有出现在待审核区，直接在通过区里面了。顺带说一下，我的邮件通知功能自部署以来就没有开启。

**Reply by Mister-Hope** [reply]
好的我们定位一下

**Reply by Mister-Hope** [reply]
我检查了代码，原则上这个是不可被绕过的。

https://github.com/walinejs/waline/blob/ecea06c201b433adb2c0e78568dcc10fc2f0b3b5/packages/server/src/controller/comment.js#L302-L308

在这里会把状态标记为未知，攻击者应该不能有机会再把它改成别的。

**Reply by Mister-Hope** [reply]
更新，我已经新建了issue，由于我未在代码中找到可以绕过的权限漏洞，将由@lizheming 二次进行检查。

**Reply by ocoke** [reply]
我收到了与 图2 格式相同的攻击

**Reply by Mister-Hope** [reply]
其实这种攻击方式怎么说呢，是个正常人也不会这么宣传自己的项目吧🤡

**Reply by Mister-Hope** [reply]
另外说实在话有几个人能无聊到四处搜博客往上贴垃圾评论的。
大部分要真想宣传的话，在知乎、掘金、博客园宣传一下就得了，谁是脑抽了么到别人的博客上去刷自己的项目...🤡这也能叫宣传？

**Reply by Moemu** [reply]
Mister-Hope: 我会将评论功能重新开启，然后等攻击者发动攻击，通过录制视频的方式看看攻击者是不是确实绕过了评论审核
